File: electit_backend/src/main/java/org/sami/electit/shared/domain/entities/Role.java
package org.sami.electit.shared.domain.entities;

import org.springframework.security.core.GrantedAuthority;

public enum Role implements GrantedAuthority {
    VOTER,
    ORGANIZER;

    @Override
    public String getAuthority() {
        return "ROLE_" + this.name();
    }
}

File: electit_frontend/lib/features/shared/domain/role.dart
enum Role {
  voter,
  organizer;

  static Role fromString(String role) {
    return switch (role) {
      'VOTER' => Role.voter,
      'ORGANIZER' => Role.organizer,
      _ => throw Exception('Unknown role: $role'),
    };
  }

  String toShortString() {
    return switch (this) {
      Role.voter => 'VOTER',
      Role.organizer => 'ORGANIZER',
    };
  }
}
